### PR TITLE
Support ingress annotations (kuma.io/ingress-public-address and kuma.…

### LINKED ISSUE #1795: Support  ingress annotations (kuma.io/ingress-public-address and kuma.io/ingress-public-port) in helm
### Summary

I'd like to provide ingress annotations (kuma.io/ingress-public-address and kuma.io/ingress-public-port) in helm

https://github.com/kumahq/kuma/blob/master/deployments/charts/kuma/templates/ingress-deployment.yaml#L23-L24

### PR DESCRIPTION
**Pull request by bdecoste**
…io/ingress-public-port) in helm (#1795)

### Summary

Support ingress annotations (kuma.io/ingress-public-address and kuma.io/ingress-public-port) in helm

### Full changelog

* [Implement ingress annotations]

### Issues resolved

Fix #1795

### CODE REVIEW

**Comment by austince** [comment]
Looks great, thanks for the contribution @bdecoste ! I think you've found almost all the gotchas, just need to run: `make generate/kumactl/install/k8s/control-plane` and commit those results as well.

**Review by austince — APPROVED** [review]
Change looks good to me, though all e2e-type tests are having seemingly unrelated issues, cc' @nickolaev: https://app.circleci.com/pipelines/github/kumahq/kuma/6825/workflows/8a43ec13-c47b-4e2d-b294-b500016c57ec/jobs/81156

I'd say we merge this, but would like to understand what's going on with the tests.

**Comment by austince** [comment]
Looks like the failing tests are part of a larger problem/ outage with Bintray's planned maintenance: https://status.bintray.com/incidents/4mzph0y8l6hk

File: deployments/charts/kuma/templates/ingress-deployment.yaml [hunk 1]
```diff
@@ -22,6 +22,11 @@ spec:
       annotations:
         kuma.io/ingress: enabled
         kuma.io/mesh: {{ .Values.ingress.mesh }}
+        {{- if .Values.ingress.annotations }}
+        {{- range $key, $value := .Values.ingress.annotations }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (nickolaev) [reply deployments/charts/kuma/templates/ingress-deployment.yaml:29]: I will merge it as it is and will let you submit another PR if you want to cover all the relevant places in the charts.

**Review by nickolaev — APPROVED** [review]

**Pull request merged**